python/revng/tupletree.py

```diff
@@ -28,6 +29,13 @@ def _create_instance(field_value, field_type):
         raise TypeError(f"Invalid type {type(field_value)}, was expecting {field_type}")
 
 
+# Hot function, hence the lru_cache
+# Called once per field for each object instantiation
+@lru_cache(maxsize=1024, typed=True)
+def get_type_hint(clazz, name):
+    return get_type_hints(clazz)[name]
```

Comment:
* `clazz`?
* Call this `get_type_hints_cached` or something like this

Reply:
`class` is a reserved keyword, so there are alternative names like `class_` or `clazz`
I checked [PEP-8](https://peps.python.org/pep-0008/#descriptive-naming-styles) and the preferred name is `class_`, so I'll be switching to that